subsys/net/ip/tcp.c

```diff
@@ -2062,7 +2065,14 @@ NET_CONN_CB(tcp_established)
 		context->tcp->fin_rcvd = 1U;
 	}
 
-	data_len = net_pkt_remaining_data(pkt);
+	if (!IS_ENABLED(CONFIG_NET_TCP_AUTO_ACCEPT) &&
```

Comment:
Based on earlier comment, I'd consider using word "accept" in this context to be confusing. This functionality is called "listen backlog". I believe we might have an option for it (or similar) already. Please consider renaming an option to not contain word "accept".